### PR TITLE
Refactor MAISI tutorial, migrate GenerativeAI import

### LINKED ISSUE #1772: Refactor MAISI Network with New Generative AI Components
There are several tasks to refactor MAISI network based on new Generative AI components and move to the MONAI/network

1. **Remove Generative Dependency**

- [x] remove generative dependency in [DiffusionModelUNetMaisi](https://github.com/Project-MONAI/MONAI/blob/dev/monai/apps/generation/maisi/networks/diffusion_model_unet_maisi.py) 
- [x]  remove generative dependency in [ControlNetMaisi](https://github.com/Project-MONAI/MONAI/blob/dev/monai/apps/generation/maisi/networks/controlnet_maisi.py)
- [x] remove generative dependency in [AutoencoderKlMaisi](https://github.com/Project-MONAI/MONAI/blob/dev/monai/apps/generation/maisi/networks/autoencoderkl_maisi.py)

2. **Ensure Compatibility of Tutorials and Scripts**

- [x] Ensure [maisi_diff_unet_training_tutorial.ipynb](https://github.com/Project-MONAI/tutorials/blob/main/generation/maisi/maisi_diff_unet_training_tutorial.ipynb) and corresponding scripts work as before.
- [x] Ensure [maisi_train_controlnet_tutorial.ipynb](https://github.com/Project-MONAI/tutorials/blob/main/generation/maisi/maisi_train_controlnet_tutorial.ipynb) and corresponding scripts work as before.
- [x] Ensure [maisi_train_vae_tutorial.ipynb](https://github.com/Project-MONAI/tutorials/blob/main/generation/maisi/maisi_train_vae_tutorial.ipynb) and corresponding scripts work as before.
- [x] Ensure [inference](https://github.com/Project-MONAI/tutorials/blob/main/generation/maisi/maisi_inference_tutorial.ipynb) work as before.
- [ ] Update state_dict to be compatible with the latest components 
- [ ] Update maisi bundle

### PR DESCRIPTION
**Pull request by Can-Zhao**
Fixes #1772 .

### Description
Refactor MAISI tutorial, migrate GenerativeAI import to monai core.
Add inference script.
Correct logging format.
VAE training notebook tested.
Diffusion training notebook tested.
ControlNet training notebook tested.
Inference notebook tested.

### Checks

- [x] Avoid including large-size files in the PR.
- [ ] Clean up long text outputs from code cells in the notebook.
- [x] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [x] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [x] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

### CODE REVIEW

**Review by mingxin-zheng — APPROVED** [review]

File: generation/maisi/maisi_train_vae_tutorial.ipynb [hunk 1]
```diff
@@ -30,20 +30,10 @@
    "execution_count": 1,
```
Comment (KumoLiu) [inline generation/maisi/maisi_train_vae_tutorial.ipynb:30]: <pre class="ql-syntax" spellcheck="false">Line #2.    !python -c "import xformers" || pip install -q xformers --index-url https://download.pytorch.org/whl/cu121
</pre><p>Can also remove xformers here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1779/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (Can-Zhao) [reply generation/maisi/maisi_train_vae_tutorial.ipynb:30]: Done. Thanks

Comment (mingxin-zheng) [reply generation/maisi/maisi_train_vae_tutorial.ipynb:30]: Hi @Can-Zhao , similarly, can we remove the same line in `generation/maisi/maisi_inference_tutorial.ipynb`?

Comment (Can-Zhao) [reply generation/maisi/maisi_train_vae_tutorial.ipynb:30]: Sorry. I forgot that. Done. Thanks! Also removed description on xformer installation in readme.

**Comment by Can-Zhao** [comment]
To pass CI, it requires monai core controlnet_maisi to remove generative dependency first.

**Comment by mingxin-zheng** [comment]
> To pass CI, it requires monai core ControlNetMaisi to remove generative dependency first.

It seems we can re-run it after this PR is merged:
https://github.com/Project-MONAI/MONAI/pull/8005

File: generation/maisi/maisi_inference_tutorial.ipynb [hunk 2]
```diff
@@ -27,15 +27,12 @@
   },
```
Comment (KumoLiu) [inline generation/maisi/maisi_inference_tutorial.ipynb:27]: <p>please help remove this line: To run this tutorial, please install&nbsp;<code>xformers</code>&nbsp;by following the&nbsp;<a href="https://github.com/facebookresearch/xformers#installing-xformers" target="_blank">official guide</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1779/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (KumoLiu) [inline generation/maisi/maisi_inference_tutorial.ipynb:27]: <pre class="ql-syntax" spellcheck="false">Line #27.    mask_generation_diffusion_unet.load_old_state_dict(checkpoint_mask_generation_diffusion_unet)
</pre><p><code>DiffusionModelUNetMaisi</code> don't have <code>load_old_state_dict</code> .</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1779/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

File: generation/maisi/maisi_train_vae_tutorial.ipynb [hunk 3]
```diff
@@ -30,20 +30,9 @@
...
```
Comment (KumoLiu) [inline generation/maisi/maisi_train_vae_tutorial.ipynb:30]: <p>please help remove this line: To run this tutorial, please install&nbsp;<code>xformers</code>&nbsp;by following the&nbsp;<a href="https://github.com/facebookresearch/xformers#installing-xformers" target="_blank">official guide</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1779/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

File: generation/maisi/configs/config_maisi.json [hunk 4]
```diff
@@ -120,11 +120,11 @@
         "dim_split": 1
     },
     "mask_generation_diffusion_def": {
-        "_target_": "monai.apps.generation.maisi.networks.diffusion_model_unet_maisi.DiffusionModelUNetMaisi",
+        "_target_": "monai.networks.nets.diffusion_model_unet.DiffusionModelUNet",
```
Comment (KumoLiu) [inline generation/maisi/configs/config_maisi.json:123]: May I know why we change the network here? cc @dongyang0122

File: generation/maisi/scripts/inference.py [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+        setattr(args, k, val)
+        print(f"{k}: {val}")
+    print("Global config variables have been loaded.")
+
+    # ## Read in configuration setting, including network definition, body region and anatomy to generate, etc.
+    #
+    # The information used for both training and inference, like network definition, is stored in "./configs/config_maisi.json". Training and inference should use the same "./configs/config_maisi.json".
+    #
+    # The information for the inference input, like body region and anatomy to generate, is stored in "./configs/config_infer.json". Please feel free to play with it.
+    # - `"num_output_samples"`: int, the number of output image/mask pairs it will generate.
```
Comment (KumoLiu) [inline generation/maisi/scripts/inference.py:130]: Perhaps we can remove these comment lines and add a link refer to the README file. Thanks.

File: generation/maisi/README.md [hunk 6]
```diff
@@ -30,16 +30,57 @@ MAISI is based on the following papers:
 ### 1. Installation
 Please refer to the [Installation of MONAI Generative Model](../README.md).
```
Comment (mingxin-zheng) [inline generation/maisi/README.md:31]: Hi @Can-Zhao , since it only depends on monai (`monai-generative` is no longer needed), shall we remove the installation note here?

File: generation/maisi/scripts/utils.py [hunk 7]
```diff
@@ -670,10 +671,11 @@ def __call__(self, img: NdarrayOrTensor):
 
 def load_autoencoder_ckpt(load_autoencoder_path):
     """
-    Load and preprocess an autoencoder checkpoint.
+    Load a state dict from an autoencoder checkpoint trained with
+    [MONAI Generative](https://github.com/Project-MONAI/GenerativeModels).
 
-    This function loads a checkpoint file and adjusts the state dict keys
-    to match the expected format for the autoencoder.
+    The loaded state dict is for AutoencoderKL in
```
[the remark this answers is not in the document]
Comment (KumoLiu) [reply generation/maisi/scripts/utils.py:677]: Like we discussed offline, I guess we can remove these load functions here and update the state dict.

File: generation/maisi/maisi_inference_tutorial.ipynb (hunk 2 above)
Comment (Can-Zhao) [reply generation/maisi/maisi_inference_tutorial.ipynb:27]: It is DiffusionModelUNet

File: generation/maisi/configs/config_maisi.json (hunk 4 above)
Comment (Can-Zhao) [reply generation/maisi/configs/config_maisi.json:123]: It was written by @guopengf . It should be DiffusionModelUNet

File: generation/maisi/README.md (hunk 6 above)
Comment (Can-Zhao) [reply generation/maisi/README.md]: Done. Thanks

File: generation/maisi/scripts/inference.py (hunk 5 above)
Comment (Can-Zhao) [reply generation/maisi/scripts/inference.py]: Done. Thanks

File: generation/maisi/maisi_train_vae_tutorial.ipynb (hunk 3 above)
Comment (Can-Zhao) [reply generation/maisi/maisi_train_vae_tutorial.ipynb]: Done. Thanks

File: generation/maisi/maisi_inference_tutorial.ipynb (hunk 2 above)
Comment (Can-Zhao) [reply generation/maisi/maisi_inference_tutorial.ipynb:25]: Done, Thanks

File: generation/maisi/scripts/utils.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (Can-Zhao) [reply generation/maisi/scripts/utils.py]: I changed docstring, please help check if it is more clear. Thanks!

[the remark this answers is not in the document]
Comment (KumoLiu) [reply generation/maisi/scripts/utils.py]: Hi @Can-Zhao, are you planning to update the state_dict in this PR, or would you prefer to create a separate PR to update it and remove the load_state_dict functions?

[the remark this answers is not in the document]
Comment (Can-Zhao) [reply generation/maisi/scripts/utils.py]: Let'e merge this PR first. It has other modifications

File: generation/maisi/configs/config_maisi.json (hunk 4 above)
Comment (guopengf) [reply generation/maisi/configs/config_maisi.json:123]: Correct, the mask generation model is not a customized diffusion unet

File: generation/maisi/scripts/utils.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (KumoLiu) [reply generation/maisi/scripts/utils.py]: Sure.

**Comment by KumoLiu** [comment]
Hi @Can-Zhao, I find there is still one place need to be updated in the maisi: https://github.com/Project-MONAI/tutorials/blob/main/generation/maisi/configs/config_maisi_diff_model.json#L3

**Comment by Can-Zhao** [comment]
> Hi @Can-Zhao, I find there is still one place need to be updated in the maisi: https://github.com/Project-MONAI/tutorials/blob/main/generation/maisi/configs/config_maisi_diff_model.json#L3

Submitted a PR for it https://github.com/Project-MONAI/tutorials/pull/1786
